plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/IcebergOptimizeHandle.java

```diff
@@ -26,111 +24,28 @@
-    private final boolean retriesEnabled;
-
-    @JsonCreator
-    public IcebergOptimizeHandle(
-            Optional<Long> snapshotId,
-            String schemaAsJson,
-            String partitionSpecAsJson,
-            List<IcebergColumnHandle> tableColumns,
-            List<TrinoSortField> sortOrder,
-            IcebergFileFormat fileFormat,
-            Map<String, String> tableStorageProperties,
-            DataSize maxScannedFileSize,
-            boolean retriesEnabled)
-    {
-        this.snapshotId = snapshotId;
-        this.schemaAsJson = requireNonNull(schemaAsJson, "schemaAsJson is null");
-        this.partitionSpecAsJson = requireNonNull(partitionSpecAsJson, "partitionSpecAsJson is null");
-        this.tableColumns = ImmutableList.copyOf(requireNonNull(tableColumns, "tableColumns is null"));
-        this.sortOrder = ImmutableList.copyOf(requireNonNull(sortOrder, "sortOrder is null"));
-        this.fileFormat = requireNonNull(fileFormat, "fileFormat is null");
-        this.tableStorageProperties = ImmutableMap.copyOf(requireNonNull(tableStorageProperties, "tableStorageProperties is null"));
-        this.maxScannedFileSize = requireNonNull(maxScannedFileSize, "maxScannedFileSize is null");
-        this.retriesEnabled = retriesEnabled;
-    }
-
-    @JsonProperty
-    public Optional<Long> getSnapshotId()
-    {
-        return snapshotId;
-    }
-
-    @JsonProperty
-    public String getSchemaAsJson()
-    {
-        return schemaAsJson;
-    }
-
-    @JsonProperty
-    public String getPartitionSpecAsJson()
-    {
-        return partitionSpecAsJson;
-    }
-
-    @JsonProperty
-    public List<IcebergColumnHandle> getTableColumns()
-    {
-        return tableColumns;
-    }
-
-    @JsonProperty
-    public List<TrinoSortField> getSortOrder()
-    {
-        return sortOrder;
-    }
-
-    @JsonProperty
-    public IcebergFileFormat getFileFormat()
-    {
-        return fileFormat;
-    }
-
-    @JsonProperty
-    public Map<String, String> getTableStorageProperties()
-    {
-        return tableStorageProperties;
-    }
-
-    @JsonProperty
-    public DataSize getMaxScannedFileSize()
-    {
-        return maxScannedFileSize;
-    }
-
-    @JsonProperty
-    public boolean isRetriesEnabled()
-    {
-        return retriesEnabled;
-    }
-
-    @Override
-    public String toString()
-    {
-        return toStringHelper(this)
-                .add("snapshotId", snapshotId)
-                .add("schemaAsJson", schemaAsJson)
-                .add("partitionSpecAsJson", partitionSpecAsJson)
-                .add("tableColumns", tableColumns)
-                .add("fileFormat", fileFormat)
-                .add("tableStorageProperties", tableStorageProperties)
-                .add("maxScannedFileSize", maxScannedFileSize)
-                .add("retriesEnabled", retriesEnabled)
-                .toString();
+    public IcebergOptimizeHandle
+    {
+        requireNonNull(schemaAsJson, "schemaAsJson is null");
```

Comment:
requireNonNull for snapshotId looks missing. Please feel free to ignore or handle in follow-up as it's pre-existing.

Reply:
let me follow-up

Reply:
https://github.com/trinodb/trino/pull/21498

Reply:
BTW -- keen eye!